### DISCUSSION TITLE
Improve the documentation for Classes

### DISCUSSION THREAD

**Discussion by nicholascar** [RDFLib/rdflib#1756] [Documentation]
In the Tern class hierarchy, these classes need better docstrings so they can be understood:

* `Node`
* `Identifier`
* `Genid` (none)
* `RDFLibGenid` (node)
* `Variable`

**Comment by aucampia** [comment]
Suggestion for `Variable` docstring here:

 - https://github.com/RDFLib/rdflib/pull/1682#issuecomment-1019009256

**Comment by nicholascar** [comment]
I wonder what `Genid` and `RDFLibGenid` are even used for? I'll have to search the codebase as I've never come across them using RDFLib!

**Comment by aucampia** [comment]
> I wonder what `Genid` and `RDFLibGenid` are even used for? I'll have to search the codebase as I've never come across them using RDFLib!

If memory serves, then one is for generating blank node identifiers and the other is not used, but I may be wrong about it and I can't recall which is which.

**Comment by edmondchuc** [comment]
> I wonder what Genid and RDFLibGenid are even used for?

They are used for skolemization and de-skolemization of blank nodes. I remember writing a brief explanation in this PR https://github.com/RDFLib/rdflib/pull/1493

**Comment by nicholascar** [comment]
Thanks for the reminder @edmondchuc!
